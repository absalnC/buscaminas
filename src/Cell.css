.cell{
	width:2em;
	height:1em;
	display:inline-block;
}