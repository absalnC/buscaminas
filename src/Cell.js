import React from "react";
import PropTypes from "prop-types";
import "./Cell.css";
class Cell extends React.Component{
	constructor(props){
		super(props);
	}

	render(){
		const color=this.props.beenPressed?(this.props.hasBomb?"red":"white"):"black";

		const st={
			border:color==="white"?"solid black 1px":"solid red 1px",
			backgroundColor:color
		}
		return (
			<div  className="cell" id={this.props.row+"-"+this.props.col} style={st} onClick={this.props.press} >
				{this.props.beenPressed?this.props.neighbours:""}
			</div>);
	}

}

Cell.propTypes={
	beenPressed:PropTypes.bool.isRequired,
	hasBomb:PropTypes.bool.isRequired,
	neighbours:PropTypes.number.isRequired,
	row:PropTypes.number.isRequired,
	col:PropTypes.number.isRequired,
	press:PropTypes.func.isRequired,

}
export default Cell;

