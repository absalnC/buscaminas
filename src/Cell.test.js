import {shallow,configure} from "enzyme";
import React from "react";
import Cell from "./Cell.js";

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

//describe("<Cell/>",()=>{
	it("should always have name formed by props.row-props.col",()=>{
		const mockFun=jest.fn();
		const el=shallow(<Cell beenPressed={false} hasBomb={true} neighbours={3} row={3} col={3} press={mockFun} />);
		expect(el.find("div").prop("name")).toBe("3-3");
	});

	it("should be black when hasNotBeenPressed and not show any content",()=>{
		const mockFun=jest.fn();
		const el=shallow(<Cell beenPressed={false} hasBomb={true} neighbours={3}  row={3} col={3} press={mockFun}/>);
		expect(el).toMatchSnapshot();
	});
	it("should show neighbours when beenPressed",()=>{
		const mockFun=jest.fn();
		const el=shallow(<Cell beenPressed={false} hasBomb={true} neighbours={3} row={3} col={3} press={mockFun}/>);
		expect(el).toMatchSnapshot();
		el.setProps({beenPressed:true,hasBomb:false,neighbours:3,row:3,col:3,press:mockFun});
		expect(el).toMatchSnapshot();
	});
	it("should be white when  beenPressed and not hasBomb",()=>{
		const mockFun=jest.fn();
		const el=shallow(<Cell beenPressed={false} hasBomb={false} neighbours={3} row={3} col={3} press={mockFun}/>);
		expect(el).toMatchSnapshot();
		el.setProps({beenPressed:true,hasBomb:false,neighbours:3,row:3,col:3,press:mockFun});
		expect(el).toMatchSnapshot();
	});
	

	it("should show switch to red when been pressed and hasBomb is true",()=>{
		const mockFun=jest.fn();
		const el=shallow(<Cell beenPressed={false} hasBomb={true} neighbours={3} row={3} col={3} press={mockFun}/>);
		expect(el).toMatchSnapshot();
		el.setProps({beenPressed:true,hasBomb:true,neighbours:3,row:3,col:3,press:mockFun});
		expect(el).toMatchSnapshot();	
	});

	it("should call props.press when is pressed",()=>{
		const mockFun=jest.fn();
		
		const el=shallow(<Cell beenPressed={false} hasBomb={true} neighbours={3} row={3} col={3} press={mockFun}/>);
		el.simulate("click");
		expect(mockFun.mock.calls.length).toBe(1);

	});
	
//})