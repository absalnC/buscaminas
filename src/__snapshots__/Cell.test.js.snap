// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be black when hasNotBeenPressed and not show any content 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    beenPressed={false}
    col={3}
    hasBomb={true}
    neighbours={3}
    press={[Function]}
    row={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "name": "3-3",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "black",
        "border": "solid red 1px",
      },
    },
    "ref": null,
    "rendered": "",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "name": "3-3",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "black",
          "border": "solid red 1px",
        },
      },
      "ref": null,
      "rendered": "",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should be white when  beenPressed and not hasBomb 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    beenPressed={false}
    col={3}
    hasBomb={false}
    neighbours={3}
    press={[Function]}
    row={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "name": "3-3",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "black",
        "border": "solid red 1px",
      },
    },
    "ref": null,
    "rendered": "",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "name": "3-3",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "black",
          "border": "solid red 1px",
        },
      },
      "ref": null,
      "rendered": "",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should be white when  beenPressed and not hasBomb 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    beenPressed={true}
    col={3}
    hasBomb={false}
    neighbours={3}
    press={[Function]}
    row={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": 3,
      "name": "3-3",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "white",
        "border": "solid black 1px",
      },
    },
    "ref": null,
    "rendered": 3,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 3,
        "name": "3-3",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "white",
          "border": "solid black 1px",
        },
      },
      "ref": null,
      "rendered": 3,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should show neighbours when beenPressed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    beenPressed={false}
    col={3}
    hasBomb={true}
    neighbours={3}
    press={[Function]}
    row={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "name": "3-3",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "black",
        "border": "solid red 1px",
      },
    },
    "ref": null,
    "rendered": "",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "name": "3-3",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "black",
          "border": "solid red 1px",
        },
      },
      "ref": null,
      "rendered": "",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should show neighbours when beenPressed 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    beenPressed={true}
    col={3}
    hasBomb={false}
    neighbours={3}
    press={[Function]}
    row={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": 3,
      "name": "3-3",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "white",
        "border": "solid black 1px",
      },
    },
    "ref": null,
    "rendered": 3,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 3,
        "name": "3-3",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "white",
          "border": "solid black 1px",
        },
      },
      "ref": null,
      "rendered": 3,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should show switch to red when been pressed and hasBomb is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    beenPressed={false}
    col={3}
    hasBomb={true}
    neighbours={3}
    press={[Function]}
    row={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "name": "3-3",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "black",
        "border": "solid red 1px",
      },
    },
    "ref": null,
    "rendered": "",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "name": "3-3",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "black",
          "border": "solid red 1px",
        },
      },
      "ref": null,
      "rendered": "",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should show switch to red when been pressed and hasBomb is true 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    beenPressed={true}
    col={3}
    hasBomb={true}
    neighbours={3}
    press={[Function]}
    row={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": 3,
      "name": "3-3",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "red",
        "border": "solid red 1px",
      },
    },
    "ref": null,
    "rendered": 3,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 3,
        "name": "3-3",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "red",
          "border": "solid red 1px",
        },
      },
      "ref": null,
      "rendered": 3,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
